### PR TITLE
Use and extend remark-preset-lint-recommended

### PR DESCRIPTION
**Pull request by nschonni**

### CODE REVIEW

**Comment by nschonni** [comment]
The other preset https://github.com/remarkjs/remark-lint/tree/master/packages/remark-preset-lint-markdown-style-guide is a more prescriptive, so I thought I'd try this first

File: index.js [hunk 1]
```diff
@@ -19,31 +21,26 @@ module.exports.plugins = [
...
   [require("remark-lint-fenced-code-marker"), "`"],
   [require("remark-lint-file-extension"), "md"],
-  [require("remark-lint-first-heading-level"), 1],
   require("remark-lint-final-definition"),
-  require("remark-lint-final-newline"),
-  require("remark-lint-hard-break-spaces"),
-  require("remark-lint-hard-break-spaces"),
+  [require("remark-lint-first-heading-level"), 1],
   [require("remark-lint-heading-style"), "atx"],
+  [require("remark-lint-list-item-bullet-indent"), false],
```
Comment (nschonni) [inline index.js]: This could be re-enabled if https://github.com/nodejs/node/pull/29488 lands

Comment (nschonni) [reply index.js]: I'll re-enable this now that the other PR landed

Comment (nschonni) [reply index.js]: This is enabled again

**Comment by nschonni** [comment]
@Trott any opinions on this?

**Comment by Trott** [comment]
> @Trott any opinions on this?

No strong opinion either way. On the one hand, I like the standardization and I like having fewer direct dependencies. On the other hand, it does mean that we have more indirect dependencies in `node_modules` and we're not using a lot of them. I'm also not sure that the disabling of rules isn't or won't be maintenance friction. Overall, I'm a +0 on this and would be inclined to land it, but I'd prefer to see what other folks think. @watiled @richardlau

**Comment by Trott** [comment]
> @watiled @richardlau

Whoops. @watilde

File: index.js [hunk 2]
```diff
@@ -19,31 +21,25 @@ module.exports.plugins = [
...
   require("remark-lint-no-file-name-consecutive-dashes"),
   require("remark-lint-no-file-name-outer-dashes"),
-  require("remark-lint-no-heading-content-indent"),
   require("remark-lint-no-heading-indent"),
-  require("remark-lint-no-inline-padding"),
+  [require("remark-lint-no-literal-urls"), false],
   require("remark-lint-no-multiple-toplevel-headings"),
   require("remark-lint-no-shell-dollars"),
-  require("remark-lint-no-shortcut-reference-image"),
+  [require("remark-lint-no-shortcut-reference-link"), false],
```
Comment (nschonni) [inline index.js]: This could be enabled with https://github.com/nodejs/node/pull/29558

File: index.js [hunk 3]
```diff
@@ -19,31 +21,25 @@ module.exports.plugins = [
...
   require("remark-lint-no-table-indentation"),
   require("remark-lint-no-tabs"),
   require("remark-lint-no-trailing-spaces"),
-  require("remark-lint-no-unused-definitions"),
+  [require("remark-lint-no-undefined-references"), false],
```
Comment (nschonni) [inline index.js]: This could be enabled with https://github.com/nodejs/node/pull/29558

**Comment by richardlau** [comment]
> 
> 
> > @Trott any opinions on this?
> 
> No strong opinion either way. On the one hand, I like the standardization and I like having fewer direct dependencies. On the other hand, it does mean that we have more indirect dependencies in `node_modules` and we're not using a lot of them. I'm also not sure that the disabling of rules isn't or won't be maintenance friction. Overall, I'm a +0 on this and would be inclined to land it, but I'd prefer to see what other folks think. @watiled @richardlau

I also have no strong opinions either way.

**Comment by Trott** [comment]
I do like how this (now, at least) means a little less configuration and a little more convention.

File: index.js (hunk 2 above)
Comment (nschonni) [reply index.js]: New PR for fixing these is https://github.com/nodejs/node/pull/29911

File: index.js (hunk 3 above)
Comment (nschonni) [reply index.js]: This is re-enabled since https://github.com/nodejs/node/pull/29809 landed

**Pull request merged**